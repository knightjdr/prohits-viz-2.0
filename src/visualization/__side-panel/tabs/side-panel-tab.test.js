import React from 'react';
import { shallow } from 'enzyme';

import SidePanelTab from './side-panel-tab';

const onClick = jest.fn();

describe('SidePanelTab', () => {
  describe('when tab is selected', () => {
    let wrapper;

    beforeAll(() => {
      wrapper = shallow(
        <SidePanelTab
          onClick={onClick}
          selectedTab="tab"
          tab="tab"
        />,
      );
    });

    it('should match snapshot', () => {
      expect(wrapper).toMatchSnapshot();
    });

    it('should have selected class', () => {
      const button = wrapper.find('button');
      expect(button.hasClass('side-panel-tab_selected')).toBeTruthy();
    });

    it('should call click handler with tab name', () => {
      onClick.mockClear();
      const button = wrapper.find('button');
      button.simulate('click');
      expect(onClick).toHaveBeenCalledWith('tab');
    });
  });

  describe('when tab is not selected', () => {
    let wrapper;

    beforeAll(() => {
      wrapper = shallow(
        <SidePanelTab
          onClick={onClick}
          selectedTab="tab"
          tab="othertab"
        />,
      );
    });

    it('should match snapshot', () => {
      expect(wrapper).toMatchSnapshot();
    });

    it('should not have selected class', () => {
      const button = wrapper.find('button');
      expect(button.hasClass('side-panel-tab_selected')).not.toBeTruthy();
    });
  });
});
