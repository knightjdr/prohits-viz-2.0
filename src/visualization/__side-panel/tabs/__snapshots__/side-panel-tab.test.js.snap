// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SidePanelTab when tab is not selected should match snapshot 1`] = `
<button
  className={null}
  onClick={[Function]}
  type="button"
>
  <FontAwesomeIcon$1
    border={false}
    className=""
    fixedWidth={false}
    flip={null}
    icon={null}
    listItem={false}
    mask={null}
    name=""
    pull={null}
    pulse={false}
    rotation={null}
    size="2x"
    spin={false}
    symbol={false}
    transform={null}
  />
</button>
`;

exports[`SidePanelTab when tab is selected should match snapshot 1`] = `
<button
  className="side-panel-tab_selected"
  onClick={[Function]}
  type="button"
>
  <FontAwesomeIcon$1
    border={false}
    className=""
    fixedWidth={false}
    flip={null}
    icon={null}
    listItem={false}
    mask={null}
    name=""
    pull={null}
    pulse={false}
    rotation={null}
    size="2x"
    spin={false}
    symbol={false}
    transform={null}
  />
</button>
`;
